pkgs/applications/window-managers/sway/default.nix

```diff
@@ -1,40 +1,62 @@
 { stdenv, fetchFromGitHub
-, cmake, pkgconfig, asciidoc, libxslt, docbook_xsl
+, cmake
+, meson, ninja
+, pkgconfig, asciidoc, libxslt, docbook_xsl
 , wayland, wlc, libxkbcommon, pcre, json_c, dbus
 , pango, cairo, libinput, libcap, pam, gdk_pixbuf, libpthreadstubs
-, libXdmcp
+, libXdmcp, wlroots, wayland-protocols, git
 , buildDocs ? true
 }:
 
-stdenv.mkDerivation rec {
-  name = "sway-${version}";
-  version = "0.15.2";
+let
+  versions = {
+    sway = {
+      version = "0.15.2";
+      sha256 = "1p9j5gv85lsgj4z28qja07dqyvqk41w6mlaflvvm9yxafx477g5n";
+      cmakeBuild = true;
+    };
 
-  src = fetchFromGitHub {
-    owner = "swaywm";
-    repo = "sway";
-    rev = version;
-    sha256 = "1p9j5gv85lsgj4z28qja07dqyvqk41w6mlaflvvm9yxafx477g5n";
+    sway-beta = {
+      version = "1.0-beta.1";
+      sha256 = "0h9kgrg9mh2acks63z72bw3lwff32pf2nb4i7i5xhd9i6l4gfnqa";
+      mesonBuild = true;
+    };
   };
 
-  nativeBuildInputs = [
-    cmake pkgconfig
-  ] ++ stdenv.lib.optional buildDocs [ asciidoc libxslt docbook_xsl ];
-  buildInputs = [
-    wayland wlc libxkbcommon pcre json_c dbus
-    pango cairo libinput libcap pam gdk_pixbuf libpthreadstubs
-    libXdmcp
-  ];
-
-  enableParallelBuilding = true;
-
-  cmakeFlags = "-DVERSION=${version} -DLD_LIBRARY_PATH=/run/opengl-driver/lib:/run/opengl-driver-32/lib";
-
-  meta = with stdenv.lib; {
-    description = "i3-compatible window manager for Wayland";
-    homepage    = http://swaywm.org;
-    license     = licenses.mit;
-    platforms   = platforms.linux;
-    maintainers = with maintainers; [ primeos ]; # Trying to keep it up-to-date.
+  common = pname: {version, sha256, cmakeBuild ? false, mesonBuild ? false}: stdenv.mkDerivation rec {
+    name = "${pname}-${version}";
+
+    src = fetchFromGitHub {
+      owner = "swaywm";
+      repo = "sway";
+      rev = version;
+      inherit sha256;
+    };
+
+    nativeBuildInputs = [
+      pkgconfig
+    ] ++ stdenv.lib.optional buildDocs [ asciidoc libxslt docbook_xsl
+    ] ++ stdenv.lib.optional cmakeBuild [ cmake
+    ] ++ stdenv.lib.optional mesonBuild [ meson ninja
+    ];
+
+
+    buildInputs = [
+      wayland wlc libxkbcommon pcre json_c dbus
+      pango cairo libinput libcap pam gdk_pixbuf libpthreadstubs
+      libXdmcp wlroots wayland-protocols git
```

Comment:
I didn't look at the source, but having `git` in `buildInputs` looks unusual for this kind of package. Why is this required?

Reply:
It fails without it (last few lines of log):

```
Library pam found: YES
Library crypt found: YES
Native dependency libsystemd found: YES 239
Dependency libelogind found: NO
Library m found: YES
Library rt found: YES
Program git found: NO
Native dependency xcb found: YES 1.13.1
Program scdoc found: NO

meson.build:130:2: ERROR:  Problem encountered: git is required to make the version string

A full log can be found at /build/source/build/meson-logs/meson-log.txt
builder for '/nix/store/phhphsml185jdmz7f13yh3ic7xpcgb4k-sway-beta-1.0-beta.1.drv' failed with exit code 1
error: build of '/nix/store/phhphsml185jdmz7f13yh3ic7xpcgb4k-sway-beta-1.0-beta.1.drv' failed
```

Reply:
Is it actually used at runtime, or can we put it in `nativeBuildInputs`?

Reply:
Oh, yeah, I think `nativeBuildInputs` would be fine as well.